src/admin_extra/service_admin_extra_accounts.erl

```diff
@@ -106,199 +104,43 @@ commands() ->
-    {ok, format_deleted_users(Users)}.
+    {Res, _} = mongoose_account_api:delete_old_users(Days),
+    Res.
 
+-spec delete_old_users_for_domain(jid:server(), integer()) ->
+    mongoose_account_api:delete_old_users().
 delete_old_users_for_domain(Domain, Days) ->
-    Users = delete_and_return_old_users(Domain, Days),
-    {ok, format_deleted_users(Users)}.
-
-delete_and_return_old_users(Domain, Days) ->
-    Users = ejabberd_auth:get_vh_registered_users(Domain),
-    delete_old_users(Days, Users).
-
--spec delete_old_users(Days, Users) -> Users when Days :: integer(),
-                                                  Users :: [jid:simple_bare_jid()].
-delete_old_users(Days, Users) ->
-    %% Convert older time
-    SecOlder = Days*24*60*60,
-
-    %% Get current time
-    TimeStampNow = erlang:system_time(second),
-
-    %% Apply the remove function to every user in the list
-    lists:filter(fun(User) ->
-                         delete_old_user(User, TimeStampNow, SecOlder)
-                 end, Users).
-
-format_deleted_users(Users) ->
-    io_lib:format("Deleted ~p users: ~p", [length(Users), Users]).
-
--spec delete_old_user(User :: jid:simple_bare_jid(),
-                      TimeStampNow :: non_neg_integer(),
-                      SecOlder :: non_neg_integer()) -> boolean().
-delete_old_user({LUser, LServer}, TimeStampNow, SecOlder) ->
-    %% Check if the user is logged
-    JID = jid:make(LUser, LServer, <<>>),
-    case ejabberd_sm:get_user_resources(JID) of
-        [] -> delete_old_user_if_nonactive_long_enough(JID, TimeStampNow, SecOlder);
-        _ -> false
-    end.
-
--spec delete_old_user_if_nonactive_long_enough(JID :: jid:jid(),
-                                               TimeStampNow :: non_neg_integer(),
-                                               SecOlder :: non_neg_integer()) -> boolean().
-delete_old_user_if_nonactive_long_enough(JID, TimeStampNow, SecOlder) ->
-    {LUser, LServer} = jid:to_lus(JID),
-    {ok, HostType} = mongoose_domain_api:get_domain_host_type(LServer),
-    case mod_last:get_last_info(HostType, LUser, LServer) of
-        {ok, TimeStamp, _Status} ->
-            %% get his age
-            Sec = TimeStampNow - TimeStamp,
-            %% If he is younger than SecOlder:
-            case Sec < SecOlder of
-                true ->
-                    %% do nothing
-                    false;
-                %% older:
-                false ->
-                    %% remove the user
-                    ejabberd_auth:remove_user(JID),
-                    true
-            end;
-        not_found ->
-            ejabberd_auth:remove_user(JID),
-            true
-    end.
+    {Res, _} = mongoose_account_api:delete_old_users_for_domain(Domain, Days),
+    Res.
 
 -spec ban_account(jid:user(), jid:server(), binary() | string()) ->
-    {ok, string()} | {error, string()}.
+    mongoose_account_api:change_password_result().
```

Comment:
Is this result type correct?

Reply:
Yes, function `ban_account` changes the password and returns a result of this operation.